### ISSUE TITLE
OpenFeign native support

### ISSUE THREAD

**Issue by driverpt** [line/armeria#3049]

**Comment by trustin** [comment]
Yeah, we'd like to provide an integration for OpenFeign like we did for Retrofit. Just didn't have time for doing that since Retrofit has been way more popular than OpenFeign internally.

**Comment by driverpt** [comment]
Do you have any Block Diagram of the `WebClient` architecture so i can try to contribute ?

**Comment by trustin** [comment]
We don't but the source code of `armeria-retrofit` is a good starting point because OpenFeign and Retrofit are quite similar to each other. Please feel free to add any questions you have while investigating. I'd be happy to answer them. :bow:
